gandiva/src/main/java/org/apache/arrow/gandiva/evaluator/JniLoader.java

```diff
@@ -69,6 +75,9 @@ private static void loadGandivaLibraryFromJar(final String tmpDir)
     final String libraryToLoad =
         LIBRARY_NAME + "/" + getNormalizedArch() + "/" + System.mapLibraryName(LIBRARY_NAME);
     final File libraryFile = moveFileFromJarToTemp(tmpDir, libraryToLoad, LIBRARY_NAME);
+    NativeLibrary.getInstance(
+        libraryFile.getAbsolutePath(),
+        Collections.singletonMap(Library.OPTION_OPEN_FLAGS, new Integer(RTLD_LAZY | RTLD_GLOBAL)));
     System.load(libraryFile.getAbsolutePath());
```

Comment:
Is this still necessary if we're using the JNA version?

Reply:
I don't think this is strictly required with JNA imho.